### PR TITLE
#[UIC-1755] Publish Y-Axis value of line chart

### PR DESCRIPTION
**Pull request by rasmi-ranjan-guavus**
UIC-1755

### CATEGORY

Choose one

- [x] Bug Fix
- [ ] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY
Publish Y-Axis value of line chart
Publish correct Y-Axis value for simple metrics, custom SQL or custom metric

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### REVIEWERS
@arpit-agarwal

### CODE REVIEW

File: superset/assets/src/visualizations/nvd3/NVD3Vis.js [hunk 1]
```diff
@@ -320,25 +319,37 @@ function nvd3Vis(element, props) {
...
             let metric;
+            let key = e.series.key;
+
             if (formData.metrics.length == 1) {
               metric = formData.metrics[0];
             } else {
-              let key = e.series.key;
               metric = formData.metrics.find((metric) => {
-                return key.includes(metric.label);
+                return metric['expressionType'] ?  key.includes(metric.label) : key.includes(metric);
```
Comment (arpit-agarwal) [inline superset/assets/src/visualizations/nvd3/NVD3Vis.js]: I liked the way you have ensured to handle string metric but can we use `hasOwnProperty` or `contructor` to see metric is a string

File: superset/assets/src/visualizations/nvd3/NVD3Vis.js [hunk 2]
```diff
@@ -320,25 +319,37 @@ function nvd3Vis(element, props) {
...
               })
             }
-            if (metric && metric.column) {
+
+            if (metric && metric.expressionType === undefined) {
```
Comment (arpit-agarwal) [inline superset/assets/src/visualizations/nvd3/NVD3Vis.js]: in what case we get expressionType as undefiend?
will this not crash when metric is a `string`

File: superset/assets/src/visualizations/nvd3/NVD3Vis.js [hunk 3]
```diff
@@ -320,25 +319,37 @@ function nvd3Vis(element, props) {
...
+              yColumn = metric;
+              yColumnLabel = metric;
+            } else if (metric && metric.expressionType === 'SIMPLE') {
               yColumn = metric.column.column_name;
```
Comment (arpit-agarwal) [inline superset/assets/src/visualizations/nvd3/NVD3Vis.js]: will we not need `metric && metric.column` check for count(*) usecase u raised earlier

File: superset/assets/src/visualizations/nvd3/NVD3Vis.js [hunk 4]
```diff
@@ -320,25 +319,37 @@ function nvd3Vis(element, props) {
...
-              let key = e.series.key;
               metric = formData.metrics.find((metric) => {
-                return key.includes(metric.label);
+                return metric['expressionType'] ?  key.includes(metric.label) : key.includes(metric);
               })
             }
-            if (metric && metric.column) {
+
+            if (metric && metric.expressionType === undefined) {
+              yColumn = metric;
```
Comment (arpit-agarwal) [inline superset/assets/src/visualizations/nvd3/NVD3Vis.js]: yColumn will become redundnat after we make `publishcolumn` changes

**Comment by arpit-agarwal** [comment]
The functionality work so far. However it is incomplete unless we update publish column support

File: superset/assets/src/visualizations/nvd3/NVD3Vis.js (hunk 3 above)
Comment (rasmi-ranjan-guavus) [reply superset/assets/src/visualizations/nvd3/NVD3Vis.js]: no, that will fall under `metric.expressionType === 'SIMPLE'` case

File: superset/assets/src/visualizations/nvd3/NVD3Vis.js (hunk 1 above)
Comment (rasmi-ranjan-guavus) [reply superset/assets/src/visualizations/nvd3/NVD3Vis.js]: updated

File: superset/assets/src/visualizations/nvd3/NVD3Vis.js (hunk 2 above)
Comment (rasmi-ranjan-guavus) [reply superset/assets/src/visualizations/nvd3/NVD3Vis.js]: in case of saved metric, expressionType will be not there

**Comment by arpit-agarwal** [comment]
We will not going ahead with this solution

**Pull request closed without merging**